source/d3d11/draw_call_tracker.cpp

```diff
@@ -1,105 +1,168 @@
 		}
 	}
 
 	bool draw_call_tracker::check_depthstencil(ID3D11DepthStencilView *depthstencil) const
 	{
 		return _counters_per_used_depthstencil.find(depthstencil) != _counters_per_used_depthstencil.end();
 	}
-	void draw_call_tracker::track_depthstencil(ID3D11DepthStencilView *depthstencil, com_ptr<ID3D11Texture2D> texture)
+
+	void draw_call_tracker::track_rendertargets(ID3D11DepthStencilView *depthstencil, com_ptr<ID3D11Texture2D> texture, UINT numviews, ID3D11RenderTargetView *const *ppRenderTargetViews)
 	{
 		assert(depthstencil != nullptr);
 
-		if (!check_depthstencil(depthstencil))
-		{
-			_counters_per_used_depthstencil.emplace(depthstencil, depthstencil_counter_info { 0, 0, texture });
+		// if depthstencil is not already being tracked, this will create it
+		_counters_per_used_depthstencil[depthstencil].depthstencil = _counters_per_used_depthstencil[depthstencil].depthstencil == nullptr ? depthstencil : _counters_per_used_depthstencil[depthstencil].depthstencil;
+		_counters_per_used_depthstencil[depthstencil].texture = _counters_per_used_depthstencil[depthstencil].texture == nullptr ? texture : _counters_per_used_depthstencil[depthstencil].texture;
+
+		for (unsigned int i = 0; i < numviews; i++) {
+			// if the rendertarget isn't being tracked, this will create it
+			_counters_per_used_depthstencil[depthstencil].additional_views[ppRenderTargetViews[i]].drawcalls += 1;
+		}
+
+	}
+
+	void draw_call_tracker::update_tracked_depthtexture(ID3D11DepthStencilView *depthstencil, com_ptr<ID3D11Texture2D> texture) {
+		auto it = _counters_per_used_depthstencil.find(depthstencil);
+		if (it != _counters_per_used_depthstencil.end()) {
+			it->second.texture = texture;
 		}
 	}
 
-	std::pair<ID3D11DepthStencilView *, ID3D11Texture2D *> draw_call_tracker::find_best_depth_stencil(UINT width, UINT height, DXGI_FORMAT format)
+	draw_call_tracker::intermediate_snapshot_info draw_call_tracker::find_best_snapshot(UINT width, UINT height, DXGI_FORMAT format)
 	{
 		const float aspect_ratio = float(width) / float(height);
 
-		depthstencil_counter_info best_info = { };
-		std::pair<ID3D11DepthStencilView *, ID3D11Texture2D *> best_match = { };
+		intermediate_snapshot_info best_snapshot;
 
-		for (const auto &it : _counters_per_used_depthstencil)
+		for (auto &[depthstencil, snapshot] : _counters_per_used_depthstencil)
 		{
-			const auto depthstencil = it.first;
-			auto &depthstencil_info = it.second;
-
-			if (depthstencil_info.drawcalls == 0 || depthstencil_info.vertices == 0)
+			if (snapshot.counter.drawcalls == 0 || snapshot.counter.vertices == 0)
 			{
 				continue;
 			}
 
-			com_ptr<ID3D11Texture2D> texture;
-
-			if (depthstencil_info.texture != nullptr)
-			{
-				texture = depthstencil_info.texture;
-			}
-			else
+			if (snapshot.texture == nullptr)
 			{
+				com_ptr<ID3D11Texture2D> texture;
 				com_ptr<ID3D11Resource> resource;
 				depthstencil->GetResource(&resource);
 
 				if (FAILED(resource->QueryInterface(&texture)))
```

Comment:
Might as well just do `if (FAILED(resource->QueryInterface(&snapshot.texture)))` and instead of using a temporary `texture` variable.